R/connect_to_etn.R

```diff
@@ -19,11 +19,28 @@
 #' con <- connect_to_etn(username = "my_username", password = "my_password")
 #' }
 connect_to_etn <- function(username, password) {
-  connection <- DBI::dbConnect(
-    odbc::odbc(),
-    "ETN",
-    uid = paste("", tolower(username), "", sep = ""),
-    pwd = paste("", password, "", sep = "")
+  tryCatch(
+    {
+      # Attempt to connect to the database with the provided credentials
+      connection <- DBI::dbConnect(
+        odbc::odbc(),
+        "ETN",
+        uid = paste("", tolower(username), "", sep = ""),
+        pwd = paste("", password, "", sep = "")
+      )
+      return(connection)
+    },
+    error = function(e) {
+        # When the database connection fails, return the error message and some
+        # directions to try again. This is usually due to a wrong password, so
+        # let's include that as a clue in the error message.
+        stop(glue::glue(e$message,
```

Comment:
The first line of the error message is difficult to understand. Would be good if it can be omitted.

![image](https://github.com/user-attachments/assets/7a0ea3ca-8c36-43c4-b4a0-35a26db87e0e)